### PR TITLE
Integrate dash tool (License Checker 1.2.3)

### PR DESCRIPTION
**Pull request by kse3hi**
- Integrate dash tool input generation as part of Licenase Checker v1.2.3
- Extend Workflow check-license with dash tool output verification

### CODE REVIEW

File: src/common/workflows/check-licenses.yml [hunk 1]
```diff
@@ -48,3 +48,25 @@ jobs:
...
+      - name: Setup Java JDK
+        uses: actions/setup-java@v1.4.4
+        with:
+          java-version: 11.0.19
+
+      - name: Run dash
+        shell: bash
+        continue-on-error: true
+        run: |
+          wget -O dash.jar "https://repo.eclipse.org/service/local/artifact/maven/redirect?r=dash-licenses&g=org.eclipse.dash&a=org.eclipse.dash.licenses&v=0.0.1-20230502.055027-834"
```
Comment (doosuu) [inline src/common/workflows/check-licenses.yml:62]: Better to use this link instead? https://github.com/eclipse/dash-licenses/releases/download/1.0.0M1/org.eclipse.dash.licenses-0.0.1-SNAPSHOT.jar

File: src/common/workflows/check-licenses.yml [hunk 2]
```diff
@@ -48,3 +48,25 @@ jobs:
...
+          java -jar dash.jar clearlydefined.input -summary DEPENDENCIES
+
+      - name: Upload dash input/output as artifacts
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: "dash.artifacts"
```
Comment (MP91) [inline src/common/workflows/check-licenses.yml:69]: I would avoid a dot "." in the name. basically it is a folder and will be downloaded as .zip.

File: src/common/workflows/check-licenses.yml (hunk 1 above)
Comment (kse3hi) [reply src/common/workflows/check-licenses.yml:62]: The link for downloading was taken from the documentation: https://github.com/eclipse/dash-licenses#get-it

It looks like they do a lot more intermediate releases compare with git releases. 
Unfortunately, they have no internal API to check the version of the JAR file.

I will check and change it, if the version on GitHub is not outdated too much and is stable.

File: src/common/workflows/check-licenses.yml (hunk 2 above)
Comment (kse3hi) [reply src/common/workflows/check-licenses.yml:69]: Agreed.

**Review by kse3hi — COMMENTED** [review]
Fix for review findings was delivered

File: src/common/workflows/check-licenses.yml [hunk 3]
```diff
@@ -48,3 +48,25 @@ jobs:
...
+          java -jar dash.jar clearlydefined.input -summary DEPENDENCIES
+
+      - name: Upload dash input/output as artifacts
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: "dash-artifacts"
+          path: |
+            clearlydefined.input
+            DEPENDENCIES
```
Comment (MP91) [inline src/common/workflows/check-licenses.yml:72]: why is the license checker not doing this directly? I mean we now have to adapt several workflows with the same 3 steps...

Comment (kse3hi) [reply src/common/workflows/check-licenses.yml:72]: My intention was to separate those activities and speed up the process for release.
I think we will consider to re-work on it later in the next PI.

**Pull request merged**